addons/website/static/tests/tours/public_widget_history.js

```diff
@@ -0,0 +1,55 @@
+/** @odoo-module **/
+
+import wTourUtils from '@website/js/tours/tour_utils';
+
+// The purpose of this tour is to check that the public widgets of snippets
+// are well destroyed when their associated HTML is removed by an history undo.
+
+wTourUtils.registerWebsitePreviewTour("test_public_widget_history", {
+    test: true,
+    url: "/",
+    edition: true,
+}, () => [
+    wTourUtils.dragNDrop({
+        name: "Instagram Page",
+    }),
+    {
+        content: "Undo the drag and drop when the instagram page is loaded",
+        trigger: "button.fa-undo:not([disabled])",
+        extra_trigger: "iframe .s_instagram_page .o_instagram_container > *",
+    },
+    {
+        content: "Wait for the instagram page to be removed",
+        trigger: "iframe body:not(:has(.s_instagram_page))",
+        run: () => {
+            // In addition, we wait for few milliseconds to be sure that the
+            // public widget is well destroyed.
+            setTimeout(() => {
+                document.body.classList.add("after_destroy");
+            }, 500);
```

Comment:
I know this kind of things is not ideal but I can't make the test running as expected (and failing when expected) without using this hack (two times actually) :grimacing:

Reply:
Yeah, we should avoid that, especially since you increase the test tour by 1 second for no good reason. Maybe you should indeed create a specific widget for this test, you would not need to do that. But it does not need to be in test_website: this is only for tests that required python things (like routes) or records we don't want to be installed for test purpose only. But you can declare a public widget only added in a test bundle in the website app. Also using the instagram snippet may not be right because it makes external call (to instagram), so we cannot use it in a normal test.